File: Solutions/2460. Apply Operations to an Array/2460.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        for i in range(len(nums)-1):
            if nums[i] == nums[i+1]:
                nums[i] *= 2
                nums[i+1] = 0
        ans = [0] * len(nums)
        i = 0
        for num in nums:
            if num != 0:
                ans[i] = num
                i += 1
        return ans
            

File: Solutions/2460. Apply Operations to an Array/2460.go
func applyOperations(nums []int) []int {
    n := len(nums)
    for i := 0; i < n - 1; i++ {
        if nums[i] == nums[i+1] {
            nums[i] *= 2
            nums[i+1] = 0
        }
    }
    ans := make([]int, n)
    i := 0
    for j := range len(nums) {
        if nums[j] != 0 {
            ans[i] = nums[j]
            i += 1
        }
    }
    return ans
}